pkg/cassandra/config_test.go

```diff
@@ -269,22 +296,67 @@ func TestCreateJsonConfig(t *testing.T) {
 			},
 			want: `{
               "cassandra-yaml": {
+				"num_tokens": 256,
                 "start_rpc": false,
                 "thrift_prepared_statements_cache_size_mb": 1
               }
             }`,
 		},
 		{
-			name:             "[4.0] start_rpc, thrift_prepared_statements_cache_size_mb",
-			cassandraVersion: "4.0",
+			name:             "[4.0.0] start_rpc, thrift_prepared_statements_cache_size_mb",
+			cassandraVersion: "4.0.0",
 			config: &api.CassandraConfig{
 				CassandraYaml: &api.CassandraYaml{
 					StartRpc:                           boolPtr(false),
 					ThriftPreparedStatementCacheSizeMb: intPtr(1),
 				},
 			},
 			want: `{
-              "cassandra-yaml": { 
+              "cassandra-yaml": {
+				"num_tokens": 16
+              }
+            }`,
+		},
+		{
+			name:             "[3.11.11] num_tokens",
+			cassandraVersion: "3.11.11",
+			config: &api.CassandraConfig{
+				CassandraYaml: &api.CassandraYaml{
+					NumTokens: intPtr(32),
+				},
+			},
+			want: `{
+              "cassandra-yaml": {
+                "num_tokens": 32
+              }
+            }`,
+		},
+		{
+			name:             "[4.0.0] num_tokens",
+			cassandraVersion: "4.0.0",
+			config: &api.CassandraConfig{
+				CassandraYaml: &api.CassandraYaml{
+					NumTokens: intPtr(32),
+				},
+			},
+			want: `{
+              "cassandra-yaml": {
+                "num_tokens": 32
+              }
+            }`,
+		},
+		{
+			name:             "[4.0.0] allocate_tokens_for_local_replication_factor",
+			cassandraVersion: "4.0.0",
+			config: &api.CassandraConfig{
+				CassandraYaml: &api.CassandraYaml{
+					AllocateTokensForLocalReplicationFactor: intPtr(5),
+				},
+			},
+			want: `{
+              "cassandra-yaml": {
+                "allocate_tokens_for_local_replication_factor": 5,
+				"num_tokens": 16
               }
             }`,
 		},
```

Comment:
Not entirely related to the changes introduced here, but I noticed that the test is not creating subtests for each test case, and it's not using testify either. I think you could update that part of the code to:

```go
for _, tc := range tests {
	t.Run(tc.name, func(t *testing.T) {
		var err error
		tc.got, err = CreateJsonConfig(tc.config, tc.cassandraVersion)
		require.NoError(t, err, "failed to create json dcConfig")
		expected, err := gabs.ParseJSON([]byte(tc.want))
		require.NoError(t, err, "failed to parse expected value")
		actual, err := gabs.ParseJSON(tc.got)
		require.NoError(t, err, "failed to parse actual value")
		assert.Equal(t, expected, actual)
	})
}
```

Reply:
Nice :)